corejs-upgrade-webpack-plugin causing problems

**Describe the bug**
If you import fetch-mock (which relies on core-js v2), its core-js imports fail.  If I hack storybook to remove the corejs-upgrade-webpack-plugin, everything seems to work fine.

Sample errors:

```
ERROR in ./node_modules/fetch-mock/es5/client.js
Module not found: Error: Cannot find module 'core-js/modules/es6.regexp.replace'
 @ ./node_modules/fetch-mock/es5/client.js 115:0-45
 @ ./src/stories/index.js
 @ ./.storybook/config.js
 @ multi ./node_modules/@storybook/core/dist/server/common/polyfills.js ./node_modules/@storybook/core/dist/server/preview/globals.js ./.storybook/config.js ./node_modules/webpack-hot-middleware/client.js?reload=true
```

**To Reproduce**
Steps to reproduce the behavior:
1. Clone https://github.com/jdelStrother/corejs-demo (a create-react-app with storybook, and an import for fetch-mock)
2. Run `yarn storybook`

**Expected behavior**
Storybook successfully builds

**System:**
 - OS: MacOS 10.14.5
 - Device: Macbook Pro 2016
 - Version: 5.1.9

**Additional context**
Fun fact: Naming your project "core-js-demo" is also a great way to break your storybook build since it matches the /core-js/ regexp that the plugin uses...

> **Fun fact:** Naming your project "core-js-demo" is also a great way to break your storybook build since it matches the /core-js/ regexp that the plugin uses...

_OMG..._
This error was _only_ happening on CI, and has been taking up most of my time these past 2 days...

My branch was called `issue/SG-16714-fix-core-js-dist-conflicts`, which on CI gets baked into the workspace path...:

```
ERR! EntryModuleNotFoundError: Entry module not found: Error: Cannot find module '/mnt/data/jenkins/workspace/ix-core-js-dist-conflicts/node_modules/@storybook/core/node_modules/html-webpack-plugin/lib/loader.js!/mnt/data/jenkins/workspace/ix-core-js-dist-conflicts/node_modules/@storybook/core/dist/server/templates/index.ejs'
```

🤦‍♂

--- a/MIGRATION.md
+++ b/MIGRATION.md
@@ -821,6 +821,8 @@ After a few iterations, this approach seems to be working. However, there are a 
 
 We'll update this section as we find more problem cases. If you have a `core-js` problem, please file an issue (preferably with a repro), and we'll do our best to get you sorted.
 
+__Update__: [corejs-upgrade-webpack-plugin](https://github.com/ndelangen/corejs-upgrade-webpack-plugin) has been removed again after running into further issues as described in [https://github.com/storybookjs/storybook/issues/7445](https://github.com/storybookjs/storybook/issues/7445).
+
 ## From version 5.0.1 to 5.0.2
 
 ### Deprecate webpack extend mode

--- a/addons/docs/package.json
+++ b/addons/docs/package.json
@@ -97,7 +97,6 @@
     "@types/util-deprecate": "^1.0.0",
     "babel-loader": "^8.0.6",
     "babel-plugin-react-docgen": "^4.1.0",
-    "corejs-upgrade-webpack-plugin": "^4.0.1",
     "cross-spawn": "^7.0.1",
     "fs-extra": "^9.0.0",
     "jest": "^25.5.2",

--- a/addons/docs/scripts/webpackDllsConfig.js
+++ b/addons/docs/scripts/webpackDllsConfig.js
@@ -1,7 +1,6 @@
 import path from 'path';
 import { ProgressPlugin, DllPlugin } from 'webpack';
 import TerserPlugin from 'terser-webpack-plugin';
-import CoreJSUpgradeWebpackPlugin from 'corejs-upgrade-webpack-plugin';
 
 const resolveLocal = (dir) => path.join(__dirname, dir);
 
@@ -60,7 +59,6 @@ export default ({ entry, provided = [] }) => ({
       path: `${out}/storybook_docs-manifest.json`,
       name: 'storybook_docs_dll',
     }),
-    new CoreJSUpgradeWebpackPlugin(),
   ],
   optimization: {
     concatenateModules: true,

--- a/lib/core/package.json
+++ b/lib/core/package.json
@@ -65,7 +65,6 @@
     "cli-table3": "0.6.0",
     "commander": "^5.0.0",
     "core-js": "^3.0.1",
-    "corejs-upgrade-webpack-plugin": "^4.0.1",
     "css-loader": "^3.5.3",
     "detect-port": "^1.3.0",
     "dotenv-webpack": "^1.7.0",

--- a/lib/core/src/server/manager/manager-webpack.config.js
+++ b/lib/core/src/server/manager/manager-webpack.config.js
@@ -4,7 +4,6 @@ import { DefinePlugin, DllReferencePlugin } from 'webpack';
 import Dotenv from 'dotenv-webpack';
 import HtmlWebpackPlugin from 'html-webpack-plugin';
 import CaseSensitivePathsPlugin from 'case-sensitive-paths-webpack-plugin';
-import CoreJSUpgradeWebpackPlugin from 'corejs-upgrade-webpack-plugin';
 import PnpWebpackPlugin from 'pnp-webpack-plugin';
 import VirtualModulePlugin from 'webpack-virtual-modules';
 
@@ -96,7 +95,6 @@ export default ({
         'process.env': stringified,
         NODE_ENV: JSON.stringify(process.env.NODE_ENV),
       }),
-      new CoreJSUpgradeWebpackPlugin({ resolveFrom: __dirname }),
     ].filter(Boolean),
     module: {
       rules: [

--- a/lib/core/src/server/preview/iframe-webpack.config.js
+++ b/lib/core/src/server/preview/iframe-webpack.config.js
@@ -6,7 +6,6 @@ import HtmlWebpackPlugin from 'html-webpack-plugin';
 import CaseSensitivePathsPlugin from 'case-sensitive-paths-webpack-plugin';
 import WatchMissingNodeModulesPlugin from 'react-dev-utils/WatchMissingNodeModulesPlugin';
 import TerserWebpackPlugin from 'terser-webpack-plugin';
-import CoreJSUpgradeWebpackPlugin from 'corejs-upgrade-webpack-plugin';
 import VirtualModulePlugin from 'webpack-virtual-modules';
 import PnpWebpackPlugin from 'pnp-webpack-plugin';
 
@@ -130,7 +129,6 @@ export default async ({
       new CaseSensitivePathsPlugin(),
       quiet ? null : new ProgressPlugin(),
       new Dotenv({ silent: true }),
-      new CoreJSUpgradeWebpackPlugin({ resolveFrom: __dirname }),
     ].filter(Boolean),
     module: {
       rules: [

--- a/lib/ui/package.json
+++ b/lib/ui/package.json
@@ -72,7 +72,6 @@
     "@testing-library/react": "^10.0.3",
     "babel-loader": "^8.0.6",
     "chromatic": "^4.0.2",
-    "corejs-upgrade-webpack-plugin": "^4.0.1",
     "enzyme": "^3.11.0",
     "flush-promises": "^1.0.2",
     "terser-webpack-plugin": "^3.0.0",

--- a/lib/ui/scripts/webpackDllsConfig.js
+++ b/lib/ui/scripts/webpackDllsConfig.js
@@ -1,7 +1,6 @@
 import path from 'path';
 import { ProgressPlugin, DllPlugin } from 'webpack';
 import TerserPlugin from 'terser-webpack-plugin';
-import CoreJSUpgradeWebpackPlugin from 'corejs-upgrade-webpack-plugin';
 
 const resolveLocal = (dir) => path.join(__dirname, dir);
 
@@ -60,7 +59,6 @@ export default ({ entry, provided = [] }) => ({
       path: `${out}/storybook_ui-manifest.json`,
       name: 'storybook_ui_dll',
     }),
-    new CoreJSUpgradeWebpackPlugin(),
   ],
   optimization: {
     concatenateModules: true,

--- a/package.json
+++ b/package.json
@@ -166,7 +166,6 @@
     "commander": "^5.1.0",
     "concurrently": "^5.2.0",
     "core-js": "^3.0.1",
-    "corejs-upgrade-webpack-plugin": "^4.0.1",
     "cross-env": "^7.0.0",
     "danger": "^10.1.1",
     "del": "^5.1.0",

